spec/controllers/webhook_controller_spec.rb

```diff
@@ -84,4 +84,38 @@ def webhook_with_secret(secret, params)
       expect(response.code).to eq '200'
     end
   end
+
+  context "with a signing secret" do
+    def webhook_with_signature(signature, params)
+      request.env['HTTP_STRIPE_SIGNATURE'] = signature
+      webhook params
+    end
+
+    def generate_signature(payload)
+      signature = Stripe::Webhook::Signature.send(:compute_signature, "#{timestamp}.#{payload}", secret)
+
+      "t=#{timestamp},v1=#{signature}"
+    end
+
+    let(:secret) { 'secret' }
+    let(:signature) { 'signature' }
+    let(:timestamp) { Time.now.to_i }
+
+    before(:each) { StripeEvent.signing_secret = secret }
+    after(:each) { StripeEvent.signing_secret = nil }
+
+    it "rejects invalid signatures" do
+      webhook_with_signature generate_signature('invalid payload'), id: 'evt_charge_succeeded'
```

Comment:
Hm, I don't think I fully understand this test. Why does this one come out as invalid with the one below valid? It seems that it should be computing a valid signature for any value that you send to `#generate_signature` shouldn't it?

Reply:
Perhaps I shouldn't use `generate_signature`? I wanted to have a spec that passes the correct format with the timestamp but with an incorrect signature. Maybe I should just pass a string directly here like so:

```ruby
webhook_with_signature "t=#{timestamp},v1=invalid", id: 'evt_charge_succeeded'
```

Perhaps a better spec would be to `"reject missing signature"` where no header is provided? Any suggestions here @rmm5t?

Reply:
I would suggest just passing in a "bogus" signature string and avoid the call to `generate_signature`.